pynamodb/compat.py

```diff
@@ -1,79 +1,8 @@
-"""
-Support for the old and outdated Python 2.6
-"""
-import logging
-import unittest
+import inspect
 
 
-class FailedToRaise(Exception):
-    pass
-
-
-class AssertRaises(object):
-    def __init__(self, exc):
-        self.expected = exc
-
-    def __enter__(self):
-        return self
-
-    def __exit__(self, exc_type, exc_value, exc_tb):
-        if exc_type is None:
-            raise Exception(
-                "{0} not raised".format(self.expected))
-
-        if not issubclass(exc_type, self.expected):
-            return False
-
-        self.exception = exc_value
-
-        return True
-
-
-class CompatTestCase(unittest.TestCase):
-
-    def assertIsNotNone(self, value, *args):
-        self.assertNotEqual(value, None, *args)
-
-    def assertIsNone(self, value, *args):
-        self.assertEqual(value, None, *args)
-
-    def assertRaises(self, excClass, callableObj=None, *args, **kwargs):
-        context = AssertRaises(excClass)
-        if callableObj is None:
-            return context
-        with context:
-            callableObj(*args, **kwargs)
-
-    def assertIsInstance(self, obj, cls, msg=None):
-        self.assertTrue(isinstance(obj, cls), msg)
-
-    def assertDictEqual(self, d1, d2, msg=None):
-        self.assertIsInstance(d1, dict, 'First argument is not a dictionary')
-        self.assertIsInstance(d2, dict, 'Second argument is not a dictionary')
-        self.assertTrue(d1 == d2, msg)
-
-    def assertListEqual(self, list1, list2, msg=None):
-        self.assertTrue(len(list1) == len(list2) and sorted(list1) == sorted(list2), msg)
-
-
-class NullHandler(logging.Handler):
-    def emit(self, record):
-        pass
-
-
-# Replace this function with inspect.getmembers() once we drop Python 2.6 support
-# see https://bugs.python.org/issue1785
-# see https://bugs.launchpad.net/zope.interface/+bug/181371
 def getmembers_issubclass(object, classinfo):
-    results = []
-    for key in dir(object):
-        try:
-            value = getattr(object, key)
-        except AttributeError:
-            continue
-
+    def predicate(value):
         value_cls = getattr(value, '__class__', None)
-        if value_cls and issubclass(value_cls, classinfo):
-            results.append((key, value))
-    results.sort()
-    return results
+        return value_cls and issubclass(value_cls, classinfo)
```

Comment:
@garrettheel do you have any idea why we essentially check `issubclass(value.__class__, classinfo)` instead of say `isinstance(value, classinfo)`?

As far as I can tell the use of `issubclass` is inherited from the initial work on the model api:
https://github.com/pynamodb/PynamoDB/blob/37bfe00/pynamodb/models.py#L116-L121

Any idea when this wouldn't be equivalent to calling `isinstance`? something, something, "old-style classes"?

On another note any idea why we do `getattr(value, '__class__', None)` instead of `value.__class__`? Looks like this was changed for "python 2 compatibility" back in https://github.com/pynamodb/PynamoDB/commit/c86e8fe . something, something, "old-style classes" again? cc/ @asottile

Reply:
from what I can tell, this function is being used to looking for attributes which are `Attribute` descriptors -- pretty sure you can more safely simplify this to `isinstance(obj, classinfo)`

`isinstance` also works fine with old-style classes, though I don't think they are at play /  relevant here

there's technically another tricking thing which isn't being handled here, the `__class__` attribute could itself be a descriptor that raises an exception -- but I don't think this is important to handle

Reply:
@asottile do you mind reviewing https://github.com/pynamodb/PynamoDB/pull/635/commits/82eef8f ?